server/controllers/article_controller.js

```diff
@@ -1,21 +1,27 @@
 import Article from '../models/article';
-import * as Group from './group_controller';
+import * as Groups from './group_controller';
 
-export const createArticle = (uri, groupid) => {
+// Should only be called when creating annotations
+export const createArticle = (uri, groupIds) => {
   const article = new Article();
   article.uri = uri;
-  article.groups.push(groupid);
-  console.log(groupid);
+  article.groups = groupIds;
   return article.save()
```

Comment:
The .then chained onto article.save() returns a promise that resolves with "result" if there's no error, and that promise is returned from createArticle. 

If you return something within a .then callback, the .then returns a promise that resolves with that value, i.e. the value becomes the argument of the next .then callback in the chain. If you throw something, it becomes the argument of the next .catch callback.